File: ToolKit/src/main/java/com/uday/android/toolkit/ui/RebootDialog.java
package com.uday.android.toolkit.ui;
import android.app.*;
import android.content.*;
import android.view.*;
import android.widget.*;
import com.uday.android.toolkit.*;

public class RebootDialog implements MenuItem.OnMenuItemClickListener
{

	private Context context;
	private RadioGroup rebootLayout;
	private String command;
	private AlertDialog dialog;
	public RebootDialog(Context context){
		this.context=context;
	}
	
	@Override
	public boolean onMenuItemClick(MenuItem p1)
	{
		// TODO: Implement this method
		if(dialog==null){
			rebootLayout=(RadioGroup)((Activity)context).getLayoutInflater().inflate(R.layout.reboot_dialog,null);
			rebootLayout.check(R.id.reboot_button);
			dialog=new AlertDialog.Builder(context).setView(rebootLayout)
							.setTitle("Advanced Reboot Menu")
							.setNegativeButton("cancel",null)
							.setPositiveButton("ok",
							new DialogInterface.OnClickListener(){
								@Override
								public void onClick(DialogInterface p1,int p2){
									p1.cancel();
									String notify="Are you sure you want to ";
									switch(rebootLayout.getCheckedRadioButtonId()){
										case R.id.power_off:
											notify=notify+"Power Off..?";
											command="setprop sys.powerctl shutdown\nsleep 3\nreboot -p";
											break;
										case R.id.reboot_button:
											notify=notify+"Reboot..?";
											command="setprop sys.powerctl reboot\nsleep 3\nreboot";
											break;
										case R.id.reboot_recovery:
											notify=notify+"reboot to recovery..?";
											command="setprop ctl.start pre-recovery\nsleep 3\nreboot recovery";
											break;
										case R.id.reboot_bootloader:
											notify=notify+"reboot to bootloader..?";
											command="reboot bootloader\n"+MainActivity.TOOL+" reboot bootloader";
											break;
										case R.id.soft_reboot:
											notify=notify+"soft reboot..? (not recomended)";
											command="setprop ctl.restart zygote\nsleep 3\n"+MainActivity.TOOL+" pkill zygote";
											break;
										case R.id.system_ui_restart:
											notify=notify+"restart SystemUi..?";
											command=MainActivity.TOOL+" pkill com.android.systemui";
											break;
									}
									
									if(context.getSharedPreferences("general",0).getBoolean("show_reboot_confirm_dialog",true))
											new DialogUtils(context).showConfirmDialog(false,0,"confirm",notify,"cancel","yes",new DialogUtils.OnClickListener(){
												@Override
												public void onClick(AlertDialog p1){
													p1.cancel();
													carryAction(command);
												}
											});
									else carryAction(command);
								}
							}).create();
			dialog.getWindow().getAttributes().windowAnimations = R.style.DialogTheme;
		}
		rebootLayout.findViewById(R.id.soft_reboot).setEnabled(context.getSharedPreferences("general",0).getBoolean("allow_soft_reboot",false));
		dialog.show();
		return true;
	}
	
	private void carryAction(String command){
		MainActivity.rootSession.addCommand(command);
	}
}

File: app/src/main/java/com/uday/android/toolkit/ui/RebootDialog.kt
package com.uday.android.toolkit.ui
import android.annotation.SuppressLint
import android.content.Context
import android.view.MenuItem
import android.widget.RadioButton
import android.widget.RadioGroup
import androidx.appcompat.app.AlertDialog
import androidx.appcompat.app.AppCompatActivity
import com.uday.android.toolkit.MainActivity
import com.uday.android.toolkit.R

class RebootDialog(private val context:Context):MenuItem.OnMenuItemClickListener {
    private var rebootLayout:RadioGroup? = null
    private var command:String? = null
    private var dialog: AlertDialog? = null

    @SuppressLint("InflateParams")
    override fun onMenuItemClick(p1:MenuItem):Boolean {
        if (dialog == null) {
            rebootLayout = (context as AppCompatActivity).layoutInflater.inflate(R.layout.reboot_dialog, null) as RadioGroup
            rebootLayout!!.check(R.id.reboot_button)
            dialog = AlertDialog.Builder(context).setView(rebootLayout)
                .setTitle("Advanced Reboot Menu")
                .setNegativeButton("cancel", null)
                .setPositiveButton("ok") {
                        p2, _ ->
                        p2.cancel()
                        var notify = "Are you sure you want to "
                        when (rebootLayout!!.checkedRadioButtonId) {
                            R.id.power_off -> {
                                notify += "Power Off..?"
                                command = "setprop sys.powerctl shutdown\nsleep 3\nreboot -p"
                            }
                            R.id.reboot_button -> {
                                notify += "Reboot..?"
                                command = "setprop sys.powerctl reboot\nsleep 3\nreboot"
                            }
                            R.id.reboot_recovery -> {
                                notify += "reboot to recovery..?"
                                command = "setprop ctl.start pre-recovery\nsleep 3\nreboot recovery"
                            }
                            R.id.reboot_bootloader -> {
                                notify += "reboot to bootloader..?"
                                command = "reboot bootloader\n" + MainActivity.TOOL + " reboot bootloader"
                            }
                            R.id.soft_reboot -> {
                                notify += "soft reboot..? (not recomended)"
                                command = "setprop ctl.restart zygote\nsleep 3\n" + MainActivity.TOOL + " pkill zygote"
                            }
                            R.id.system_ui_restart -> {
                                notify += "restart SystemUi..?"
                                command = MainActivity.TOOL!! + " pkill com.android.systemui"
                            }
                        }

                        if (context.getSharedPreferences("general", 0).getBoolean("show_reboot_confirm_dialog", true))
                            DialogUtils(context).showConfirmDialog(false, 0, "confirm", notify, "cancel", "yes", object:DialogUtils.OnClickListener {
                                override fun onClick(p1: AlertDialog?) {
                                    p1!!.cancel()
                                    carryAction(command)
                                }
                            })
                        else carryAction(command)
                }.create()
            dialog!!.window!!.attributes.windowAnimations = R.style.DialogTheme
        }
        rebootLayout!!.findViewById<RadioButton>(R.id.soft_reboot).isEnabled = context.getSharedPreferences("general", 0).getBoolean("allow_soft_reboot", false)
        dialog!!.show()
        return true
    }

    private fun carryAction(command:String?) {
        MainActivity.rootSession!!.addCommand(command)
    }
}
